ext/strings_test.go

```diff
@@ -329,3 +330,229 @@ func TestStringsWithExtension(t *testing.T) {
+		},
+		{
+			name:           "substution inside escaped percent signs",
+			format:         "%%%s%%",
+			formatArgs:     `"text"`,
+			expectedOutput: "%text%",
+		},
+		{
+			name:           "substitution with one escaped percent sign on the right",
+			format:         "%s%%",
+			formatArgs:     `"percent on the right"`,
+			expectedOutput: "percent on the right%",
+		},
+		{
+			name:           "substitution with one escaped percent sign on the left",
+			format:         "%%%s",
+			formatArgs:     `"percent on the left"`,
+			expectedOutput: "%percent on the left",
+		},
+		{
+			name:           "multiple substitutions",
+			format:         "%d %d %d, %s %s %s, %d %d %d, %s %s %s",
+			formatArgs:     `1, 2, 3, "A", "B", "C", 4, 5, 6, "D", "E", "F"`,
+			expectedOutput: "1 2 3, A B C, 4 5 6, D E F",
+		},
+		{
+			name:           "percent sign escape sequence support",
+			format:         "\u0025\u0025escaped \u0025s\u0025\u0025",
+			formatArgs:     `"percent"`,
+			expectedOutput: "%escaped percent%",
+		},
+		{
+			name:           "fixed point formatting clause",
+			format:         "%.3f",
+			formatArgs:     "1.2345",
+			expectedOutput: "1.234",
+		},
+		{
+			name:           "binary formatting clause",
+			format:         "this is 5 in binary: %b",
+			formatArgs:     "5",
+			expectedOutput: "this is 5 in binary: 101",
+		},
+		{
+			name:           "octal formatting clause",
+			format:         "%o",
+			formatArgs:     "11",
+			expectedOutput: "13",
+		},
+		{
+			name:           "lowercase hexadecimal formatting clause",
+			format:         "%h is 20 in hexadecimal",
+			formatArgs:     "30",
+			expectedOutput: "1e is 20 in hexadecimal",
+		},
+		{
+			name:           "uppercase hexadecimal formatting clause",
+			format:         "%H is 20 in hexadecimal",
+			formatArgs:     "30",
+			expectedOutput: "1E is 20 in hexadecimal",
+		},
+		{
+			name:           "string support with hexadecimal formatting clause",
+			format:         "%h",
+			formatArgs:     `"Hello world!"`,
+			expectedOutput: "48656c6c6f20776f726c6421",
+		},
+		{
+			name:           "lowercase scientific notation formatting clause",
+			format:         "%.6e",
+			formatArgs:     "1052.032911275",
+			expectedOutput: "1.052033\u202f×\u202f10⁰³",
+		},
+		{
+			name:           "uppercase scientific notation formatting clause",
+			format:         "%.6E",
+			formatArgs:     "1052.032911275",
+			expectedOutput: "1.052033\u202f×\u202f10⁰³",
+		},
+		{
+			name:           "locale support",
+			format:         "%.3f",
+			formatArgs:     "3.14",
+			locale:         "fr_FR",
+			expectedOutput: "3,140",
+		},
+		{
+			name:           "default precision for fixed-point clause",
+			format:         "%f",
+			formatArgs:     "2.71828",
+			expectedOutput: "2.718280",
+		},
+		{
+			name:           "default precision for scientific notation",
+			format:         "%e",
+			formatArgs:     "2.71828",
+			expectedOutput: "2.718280\u202f×\u202f10⁰⁰",
```

Comment:
What's going on with this output?  I was not expecting to see any escape sequences. Go just outputs `2.718280e+00`

Reply:
Swap "\u202f" in these examples with the unicode equivalent? Right now this output has a mix of escaped unicode and actual unicode.

Reply:
I've swapped it over, though I think it's worth considering that `\u202f` is a [narrow space](https://codepoints.net/U+202F) and could possibly get confused with a normal/ASCII space inside the unit tests.

Reply:
Good point. Maybe show a test case of both? It seems beneficial to be able to see both how it will _look_ and what it _really is_ in the tests.

Reply:
Right, I've reverted some of the test cases to use the escape sequence. It's also worth pointing out that that is not a regular ASCII "x" in there, but a [multiplication sign](https://codepoints.net/U+00D7?lang=en). Github's font makes it relatively clear, but I thought I'd point it out just in case. I've also added a comment making that fact about the multiplication sign, as well.

Reply:
In the unicode escaped case, escape this too, and the superscripts at end as well. Then the side-by-side of the test cases will highlight all the unicode a bit better?

Reply:
(marking this as resolved since the supercripts/multiplication signs have both been escaped in the unit tests with escaped spaces)